python.d/springboot.chart.py

```diff
@@ -0,0 +1,138 @@
+# Author: Wing924
+
+import json
+from bases.FrameworkServices.UrlService import UrlService
+
+# default module values (can be overridden per job in `config`)
+# update_every = 2
+priority = 60000
+retries = 60
+
+
+DEFAULT_ORDER = ['response_code', 'threads', 'gc_time', 'gc_ope', 'heap']
+
+DEFAULT_CHARTS = {
+    'response_code': {
+        'options': [None, "Response Codes", "requests/s", "response", "springboot.response_code", "stacked"],
+        'lines': [
+            ["resp.other",  'Other', 'incremental'],
+            ["resp.1xx",    '1xx',   'incremental'],
+            ["resp.2xx",    '2xx',   'incremental'],
+            ["resp.3xx",    '3xx',   'incremental'],
+            ["resp.4xx",    '4xx',   'incremental'],
+            ["resp.5xx",    '5xx',   'incremental'],
+        ]},
+    'threads': {
+        'options': [None, "Threads", "current threads", "threads", "springboot.threads", "area"],
+        'lines': [
+            ["threads.daemon", 'daemon', 'absolute'],
+            ["threads", 'total', 'absolute'],
+        ]},
+    'gc_time': {
+        'options': [None, "GC Time", "milliseconds", "garbage collection", "springboot.gc_time", "area"],
+        'lines': [
+            ["gc.copy.time",                'Copy',                 'incremental'],
+            ["gc.marksweepcompact.time",    'MarkSweepCompact',     'incremental'],
+            ["gc.parnew.time",              'ParNew',               'incremental'],
+            ["gc.concurrentmarksweep.time", 'ConcurrentMarkSweep',  'incremental'],
+            ["gc.ps_scavenge.time",         'PS Scavenge',          'incremental'],
+            ["gc.ps_marksweep.time",        'PS MarkSweep',         'incremental'],
+            ["gc.g1_young_generation.time", 'G1 Young Generation',  'incremental'],
+            ["gc.g1_old_generation.time",   'G1 Old Generation',    'incremental'],
+        ]},
+    'gc_ope': {
+        'options': [None, "GC Operations", "operations/s", "garbage collection", "springboot.gc_ope", "area"],
+        'lines': [
+            ["gc.copy.count",                'Copy',                 'incremental'],
+            ["gc.marksweepcompact.count",    'MarkSweepCompact',     'incremental'],
+            ["gc.parnew.count",              'ParNew',               'incremental'],
+            ["gc.concurrentmarksweep.count", 'ConcurrentMarkSweep',  'incremental'],
+            ["gc.ps_scavenge.count",         'PS Scavenge',          'incremental'],
+            ["gc.ps_marksweep.count",        'PS MarkSweep',         'incremental'],
+            ["gc.g1_young_generation.count", 'G1 Young Generation',  'incremental'],
+            ["gc.g1_old_generation.count",   'G1 Old Generation',    'incremental'],
+        ]},
+    'heap': {
+        'options': [None, "Heap Memory Usage", "KB", "heap memory", "springboot.heap", "area"],
+        'lines': [
+            ["heap.committed", 'committed', "absolute"],
+            ["heap.used", 'used', "absolute"],
```

Comment:
please replace dots with underscores in dimension ids

Reply:
The dots in id come from spring actuator output (the data source).
This is sample output:
```
{  
   "mem":128960,
   "mem.free":10627,
   "processors":1,
   "instance.uptime":243514122,
   "uptime":243523907,
   "systemload.average":0.0,
   "heap.committed":59876,
   "heap.init":61440,
   "heap.used":49248,
   "heap":936448,
   "nonheap.committed":70464,
   "nonheap.init":2496,
   "nonheap.used":69085,
   "nonheap":0,
   "threads.peak":29,
   "threads.daemon":26,
   "threads.totalStarted":34,
   "threads":29,
   "classes":7863,
   "classes.loaded":7863,
   "classes.unloaded":0,
   "gc.copy.count":83,
   "gc.copy.time":412,
   "gc.marksweepcompact.count":2,
   "gc.marksweepcompact.time":173,
   "httpsessions.max":-1,
   "httpsessions.active":0,
   "gauge.response.artist.code.root":134.0,
   "gauge.response.artist.code":68.0,
   "counter.status.200.artist.code.root":73,
   "counter.status.302.artist.code":6,
   "counter.status.302.artist.code.root":13,
   "counter.status.200.artist.code":55
}
```

Reply:
Well, still

> please replace dots with underscores in dimension id

this is needed for backends